src/acom_music_box/conditions.py

```diff
@@ -211,24 +178,26 @@ def read_initial_rates_from_file(self, file_path, reaction_list):
             list: A list where each element represents the initial rate of a reaction.
         """
 
-        reaction_rates = []
-
-        with open(file_path, 'r') as csv_file:
-            initial_conditions = list(csv.reader(csv_file))
-
-            if (len(initial_conditions) > 1):
-                # The first row of the CSV contains headers
-                headers = initial_conditions[0]
-
-                # The second row of the CSV contains rates
-                rates = initial_conditions[1]
-
-                for reaction_rate, rate in zip(headers, rates):
-                    type, name, *rest = reaction_rate.split('.')
-                    for reaction in reaction_list.reactions:
-                        if reaction.name == name and reaction.short_type() == type:
-                            reaction_rates.append(ReactionRate(reaction, rate))
-
+        reaction_rates = {}
+
+        df = pd.read_csv(file_path)
+        rows, _ = df.shape
+        if rows > 1:
+            raise ValueError(f'Initial conditions file ({file_path}) may only have one row of data. There are {rows} rows present.')
+        for key in df.columns:
+            parts = key.split('.')
+            reaction_type, label = None, None
+            if len(parts) == 3:
+                reaction_type, label, units = parts
+            elif len(parts) == 2:
+                reaction_type, label = parts
+            else:
+                logger.error(f"Unexpected format in key: {key}")
+            rate_name = f'{reaction_type}.{label}'
+            if rate_name in reaction_rates:
+                raise ValueError(f"Duplicate reaction rate found: {rate_name}")
+            reaction_rates[rate_name] = df.iloc[0][key]
+        
```

Comment:
What is the reason to add unexpected-format keys into `reaction rates` dict? My impression based on the logging level is that it's not valid and should be thrown away